xclim/indices.py

```diff
@@ -870,3 +871,164 @@ def percentile_doy(arr, window=5, per=.1):
+        else:
+            output = xr.concat((output, wd), dim='time', coords=['tags'])
+
+    # set units, and name attributes
+    output = output.rename('wet_days')
+    output['units'] = 'days'
+    output['name'] = 'wet_days'
+    output['long_name'] = 'number of wet days'
+    output['details'] = 'wet day defined as a day with pr >= {:5.2f} mm'.format(pr_min)
+
+    return output
+
+
+def daily_intensity(pr, pr_min=1., freq='YS', skipna=False):
+    r"""daily intensity
+
+    compute simple daily intensity index by averaging the precipitation over wet_days
+    for a given period
+
+    Parameters
+    ----------
+    pr : xarray.DataArray
+      Daily precipitation [mm]
+    pr_min : float
+      precipitation value over which a day is considered wet
+    freq : str, optional
+      Resampling frequency defining the periods
+      defined in http://pandas.pydata.org/pandas-docs/stable/timeseries.html#resampling.
+    skipna : str, optional
+      if True, NaN values are ignored
+      if False NaN values are expanded
+      This is used for the computation of the sum of wet days. See
+      http://xarray.pydata.org/en/stable/generated/xarray.DataArray.sum.html for details
+
+    Returns
+    -------
+    xarray.DataArray
+      The average precipitation over wet days for each period
+
+    Notes
+    -----
+
+    Examples
+    --------
+    The following would compute for each grid cell of file `pr.day.nc` the average
+    precipitation fallen over days with precipitation >= 5 mm at seasonal
+    frequency, ie DJF, MAM, JJA, SON, DJF, etc.
+
+    >>> pr = xr.open_dataset('pr.day.nc')
+    >>> daily_int = daily_intensity(pr, pr_min = 5., freq="QS-DEC")
+
+    """
+
+    # group the data according to the frequency
+    groups = daily_downsampler(pr, freq=freq).groups
+
+    # split data according to frequency
+    for n_t, tag in enumerate(groups):
+        # get data from each group
+        g = groups[tag]
+        prg = pr[g]
+
+        # put precip null on non-wet days
+        prg_wd = xr.where(prg < pr_min, 0, prg)
+
+        # sum pr over wetdays
+        prg_wd_sum = prg_wd.sum(dim='time', skipna=skipna)
+
+        # find number of wet days
+        wd = wet_days(prg, pr_min=pr_min, freq=freq, skipna=skipna)
+        wd = wd.isel(time=0)
+
+        # compute average over wet days
+        di = prg_wd_sum / wd
+
+        # keep tag and start time of group and add time dim
+        di.coords['tags'] = tag
+        di.coords['time'] = prg.time.values[0]
+
+        # accumulate the results in output DataArray
+        if n_t == 0:
+            output = di
+        else:
+            output = xr.concat((output, di), dim='time', coords=['tags'])
+
+    # set units, and name attributes
+    output = output.rename('daily_intensity')
+    output['name'] = 'daily_intensity'
+    output['long_name'] = 'daily intensity over wet days'
+    output['details'] = 'wet day defined as a day with pr >= {:5.2f} mm'.format(pr_min)
```

Comment:
Même affaire, je pense que qqch comme

```
v = (pr - pr_min).clip(min=0)

n = ((v > 0) * 1).resample(time=freq).sum(dim='time')
s = v.resample(time=freq, skipna=skipna).sum(dim='time')
return s/n
```
va marcher, non?

Reply:
Encore là, faut regarder les nans et encore là, je ne vois pas le lien entre ton code et le code en vert.

Je vais regarder ça et je te reviens ... dans qqs jours probablement

Reply:
Pour le code en vert, j'ai juste cliqué au hasard dans ta fonction, c'est tout le code qui peut être remplacé. 
D'accord que ça ne rèlge pas la question des NaNs, mais je pense qu'on doit avoir une approche plug-in.

Reply:
J'ai refait les indices et les tests et mis la branche à jour.